packages/rum/src/domain/record/serialization/serializeNode.ts

```diff
@@ -48,45 +65,54 @@ export function generateNextId(): number {
   return _nextId++
 }
 
-export function serializeChildNodes(node: Node, options: SerializeOptions): SerializedNodeWithId[] {
+export function serializeChildNodes(
+  node: Node,
+  options: SerializeOptions,
+  cache?: SerializedNodeCache
+): SerializedNodeWithId[] {
   const result: SerializedNodeWithId[] = []
   forEachChildNodes(node, (childNode) => {
-    const serializedChildNode = serializeNodeWithId(childNode, options)
+    const serializedChildNode = serializeNodeWithId(childNode, options, cache)
     if (serializedChildNode) {
       result.push(serializedChildNode)
     }
   })
   return result
 }
 
-function serializeNode(node: Node, options: SerializeOptions): SerializedNode | undefined {
+function serializeNode(node: Node, options: SerializeOptions, cache?: SerializedNodeCache): SerializedNode | undefined {
   switch (node.nodeType) {
     case node.DOCUMENT_NODE:
-      return serializeDocumentNode(node as Document, options)
+      return serializeDocumentNode(node as Document, options, cache)
     case node.DOCUMENT_FRAGMENT_NODE:
-      return serializeDocumentFragmentNode(node as DocumentFragment, options)
+      return serializeDocumentFragmentNode(node as DocumentFragment, options, cache)
```

Comment:
**💬 suggestion:** ‏to avoid passing the `cache` everywhere, you could add it as a `SerializationContext` property (similarly to `shadowRootController`)

Reply:
Done!